main.keyvault.tf

```diff
@@ -20,7 +23,7 @@ module "avm_res_keyvault_vault" {
       inherit_lock                    = value.inherit_lock
     }
   } : {}
-  tags = var.tags
+  tags = var.key_vault.tags == null ? var.tags : var.key_vault.tags == {} ? {} : var.key_vault.tags
```

Comment:
Again the same comment about merging tags or leving than out of the object completly. I am more in favour of the later.